dist/node/pusher.js

```diff
@@ -344,11 +344,43 @@ module.exports =
 	    },
 	    createXHR: function () {
 	        var Constructor = this.getXHRAPI();
-	        return new Constructor();
+			if (process.env.HTTP_PROXY_HOST) {
+	        	return new Constructor({proxy: {host: process.env.HTTP_PROXY_HOST, port: process.env.HTTP_PROXY_PORT}});
+			} else {
+	        	return new Constructor();
+			}
 	    },
 	    createWebSocket: function (url) {
 	        var Constructor = this.getWebSocketAPI();
-	        return new Constructor(url);
+
+	        /* manually editing compiled js code to solve my proxy problem
+
+	        if HTTP_PROXY or http_proxy is present in the environment, pass that info down to the websocket
+	        if WS_PROXY_* vars are present in the environment, pass those down too
+	        the WS_PROXY_* vars correspond to faye-websockets WebSocket proxy options
+	        https://github.com/faye/faye-websocket-node#using-the-websocket-client
+
+	        WS_PROXY - treated exactly the same as HTTP_PROXY and http_proxy
+	        WS_PROXY_HEADERS_JSON - JSON.stringified HTTP headers
+	        WS_PROXY_TLS_CERT - the contents of the tls cert (not a path to file)*/
+			var origin = process.env.WS_PROXY || process.env.HTTP_PROXY || process.env.http_proxy;
+
+	        if (origin) {
+	        	// origin is required for proxy config
+				var options = {proxy: {origin: origin}};
+
+				if (process.env.WS_PROXY_HEADERS_JSON) {
+					options.proxy.headers = JSON.parse(process.env.WS_PROXY_HEADERS_JSON);
+				}
+
+				if (process.env.WS_PROXY_TLS_CERT) {
+					options.proxy.tls = process.env.WS_PROXY_TLS_CERT;
+				}
+
+				return new Constructor(url, [], options);
```

Comment:
I'm confused here. I think this is doing `new WebSocket(url, [], options)`, but this call doesn't look compatible with [the `WebSocket` API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket), which doesn't take an `options` object. @leesio